plugins/out_stackdriver/stackdriver.c

```diff
@@ -277,6 +277,139 @@ static char *get_google_token(struct flb_stackdriver *ctx)
     return ctx->o->access_token;
 }
 
+static int process_local_resource_id(const void *data, size_t bytes,
+                                     struct flb_stackdriver *ctx, char *type)
+{
+    int i;
+    int len;
+    int ret = 0;
+    int flag = 0;
+    size_t off = 0;
+    msgpack_object k;
+    msgpack_object v;
+    msgpack_object root;
+    msgpack_object_map map;
+    msgpack_unpacked result;
+    char *local_resource_id;
+    char *ptr;
+    const char delim[] = ".";
+
+    msgpack_unpacked_init(&result);
+    while (msgpack_unpack_next(&result, data, bytes, &off) == MSGPACK_UNPACK_SUCCESS && !flag) {
+        root = result.data;
+        flag = 0;
+
+        if (root.type != MSGPACK_OBJECT_ARRAY ||
+            root.via.array.size != 2 ||
+            root.via.array.ptr[1].type != MSGPACK_OBJECT_MAP) {
+            flb_plg_warn(ctx->ins, "unexpected record format");
```

Comment:
nit: this looks like a common check for fluent bit's record format in msgpack form. maybe we can extract this function out and later use it as a common util function.

Reply:
updated.